channels/views/channels.py

```diff
@@ -79,8 +80,20 @@ def get_serializer_context(self):
 
     def get_object(self):
         """Get channel referenced by API"""
-        api = Api(user=self.request.user)
-        return api.get_channel(self.kwargs["channel_name"])
+        return self.request.channel_api.get_channel(self.kwargs["channel_name"])
+
+    def check_permissions(self, request):
+        """
+        Overridden base method. Deny permission if the user is not a superuser and is attempting to
+        update the channel title.
+        """
+        if (
+            request.method == "PATCH"
```

Comment:
Include `PUT` here too.  If I go to `http://od.odl.local:8063/api/v0/channels/<channel_name>/` and use the form I can still change the title as a non-superuser.